src/common/datomish/util/deque.clj

```diff
@@ -0,0 +1,24 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+(ns datomish.util.deque)
+
+(defn deque
+  "Return a new, empty deque."
+  []
+  (ref []))
```

Comment:
I think you want an `atom` here. There's no reason to use a `ref` unless you need to transactionally coordinate multiple `ref`s.